packages/web-components/src/rating-display/rating-display.stories.ts

```diff
@@ -112,6 +131,13 @@ export const Count: Story<FluentRatingDisplay> = renderComponent(html<StoryArgs<
   <fluent-rating-display value="4" count="3391"></fluent-rating-display>
 `);
 
+export const Icon: Story<FluentRatingDisplay> = renderComponent(html<StoryArgs<FluentRatingDisplay>>`
+  <fluent-rating-display
+    value="3.8"
+    icon='<svg><path d="M10 2C5.58172 2 2 5.58172 2 10C2 14.4183 5.58172 18 10 18C14.4183 18 18 14.4183 18 10C18 5.58172 14.4183 2 10 2Z" /></svg>'
+  ></fluent-rating-display>
```

Comment:
Would this create potential security risk like `<fluent-rating-display icon="<script>alert(1)</script>"></fluent-rating-display>`? Is it possible to use a slot to pass in the custom SVG icon? Or using `<template>` like:

```
<fluent-rating-display>
  <template><svg>...</svg></template>
</fluent-rating-display>
```

Reply:
You're right, a script tag can be injected and executed, good catch. But using a slot or template also won't protect us because we'll still need to grab the SVG from the slot/template and use it to dynamically render the correct number of icons. The safest way I can think of is to sanitize the SVG before inserting it to the DOM. Or, if we make it so that the component only accepts a set of predefined icons. Any thoughts?

Reply:
With slot, maybe you can use:

```
<slot ${slotted({ property: ʻ...ʻ, filter: node => node.nodeName === ʻSVGʻ })></slot>
```

Reply:
Even if the slotted element is restricted to svg, script tags can still be added inside the svg element. It also won't prevent a valid svg element to run some code with something like `<svg onload=alert("!!!")>`.

Reply:
What if we use slot and only render the icons when the component is first connected? That way, if the SVG value got modified after the initial render, it won't be added to the DOM and any code won't get executed. Although, this approach still requires us to trust that the initial SVG provided is good.

Sanitizing the SVG is probably still the safest, but it does increase the bundle size.

Reply:
FAST has mechanisms for DOM sanitation, specifically when passing strings - with that said, I don't think passing the SVG as an attribute makes sense. Checkout the inline template directive.

Reply:
Based on the discussion we had separately, I've made the change to add a slot named "icon" where a custom SVG icon can be passed, and to render the icons using `staticallyCompose()`. Additionally, the icon slot is restricted to SVGs and the default star icon is not added to the DOM if a custom icon is provided.